### ISSUE TITLE
Testnet syncronisation stops working after a few minutes, no nodes connected anymore

### ISSUE THREAD

**Issue by corck** [CityOfZion/neo-python#937]
### Current behavior
When starting testnet after a short while the connection to all nodes drops with errors in the console, therefore syncing testnet is practically broken. Syncing currently would take a few hours or a full day and you would need to permanently monitor the connection state and restart neo-python to get new node connection again. I tried this, but now switching to neo-cli, as its too cumbersome.

### Expected behavior
Neo-python should trying reconnect to nodes if all nodes dropped.  

### How to reproduce

Startup neo-python in testmode. Wait for 10-30 minutes. All nodes are lost.

### Your environment

Let us know in what environment you're running into the issue:

- OS: Ubuntu 
- neo-python version: neo-python v0.8.4
- Python version: Python 3.7.1

**Comment by ixje** [comment]
Sorry to hear that. There actually is a loop in the `0.8.4` version that monitors for 0 peers
https://github.com/CityOfZion/neo-python/blob/fe90f62e123d720d4281c79af0598d9df9e776fb/neo/Network/NodeLeader.py#L446

which should run every 10 seconds
https://github.com/CityOfZion/neo-python/blob/fe90f62e123d720d4281c79af0598d9df9e776fb/neo/Network/NodeLeader.py#L103-L105

but according to your experience it does not. Can you by any chance share the error(s) you're seeing? Perhaps there's a fix already known for it.

**Comment by corck** [comment]
Unfortunately it does not.

thats the error I think gets thrown for every node that gets disconnected, I think we discussed that in chat before.

```
Unhandled error in Deferred:                                                                                                                                                                                       
                                                                                                                                                                                                                   
Traceback (most recent call last):                                                                                                                                                                                 
  File "/root/neo-python/venv/lib/python3.7/site-packages/twisted/internet/base.py", line 1152, in connectionFailed                                                                                                
    self.factory.clientConnectionFailed(self, reason)                                                                                                                                                              
  File "/root/neo-python/venv/lib/python3.7/site-packages/twisted/internet/endpoints.py", line 263, in clientConnectionFailed                                                                                      
    self._onConnection.errback(reason)                                                                                                                                                                             
  File "/root/neo-python/venv/lib/python3.7/site-packages/twisted/internet/defer.py", line 501, in errback                                                                                                         
    self._startRunCallbacks(fail)                                                                                                                                                                                  
  File "/root/neo-python/venv/lib/python3.7/site-packages/twisted/internet/defer.py", line 568, in _startRunCallbacks                                                                                              
    self._runCallbacks()                                                                                                                                                                                           
--- <exception caught here> ---                                                                                                                                                                                    
  File "/root/neo-python/venv/lib/python3.7/site-packages/twisted/internet/defer.py", line 654, in _runCallbacks                                                                                                   
    current.result = callback(current.result, *args, **kw)                                                                                                                                                         
  File "/root/neo-python/neo/Network/NodeLeader.py", line 693, in clientConnectionFailed                                                                                                                           
    logger.debug(f"Failed connecting to {address} {ce.args[0].value}")                                                                                                                                             
builtins.AttributeError: 'str' object has no attribute 'value' 

@ixje 
```

**Comment by ixje** [comment]
I think that might be solved by https://github.com/CityOfZion/neo-python/pull/926
I can't tell for sure if this exception is causing the reactor to stop therefore breaking the recovery logic. That could be worth trying, but I can understand your need for stability. I hope we can soon bring that back.

**Comment by corck** [comment]
I'll switched to the latest development and check if I experience the same.

**Comment by corck** [comment]
@ixje synced now on two of my servers using development branch and both are now syncing stable, no errors in the console anymore, nodes connection stays alive. 
Will close this one in ancipitation of #926 being merged in a new release.

**Comment by ixje** [comment]
Thanks for giving it a try. One of the items I'm waiting on before the next release is the NEO Team to finalise their VM updates (the last code showed differences in storage on the testnet). I want to make sure `neo-python`'s VM matches their "stable" version.
